code/modules/atmospherics/FEA_turf_tile.dm

```diff
@@ -390,6 +393,18 @@ var/global/list/turf/hotly_processed_turfs = list()
 			item.temperature_expose(src.air, src.air.temperature, CELL_VOLUME)
 		temperature_expose(src.air, src.air.temperature, CELL_VOLUME)
 
+	if(src.air.radgas >= RADGAS_MINIMUM_CONTAMINATION_MOLES && !ON_COOLDOWN(src, "radgas_contaminate", RADGAS_CONTAMINATION_COOLDOWN)) //if fallout is in the air, contaminate objects on this tile and consume radgas
+		for(var/atom/movable/item in src)
+			if(isintangible(item) || isobserver(item) || istype(item, /obj/overlay) || istype(item, /obj/effects))
+				continue
+			var/datum/component/radioactive/R = item.GetComponent(/datum/component/radioactive)
+			if(R?.radStrength > RADGAS_MAXIMUM_CONTAMINATION)
+				continue
```

Comment:
`GetComponent()` 😬 
Try to use signals for interacting with components when at all possible

Reply:
I assume by this you mean I should add a signal like `COMSIG_GET_RADIOACTIVITY` and use that instead?